src/rgw/rgw_amqp.cc

```diff
@@ -395,11 +404,46 @@ connection_ptr_t& create_connection(connection_ptr_t& conn, const amqp_connectio
   ConnectionCleaner state_guard(state);
 
   // create and open socket
-  auto socket = amqp_tcp_socket_new(state);
+  amqp_socket_t *socket = nullptr;
+  if (info.ssl) {
+    socket = amqp_ssl_socket_new(state);
+#if AMQP_VERSION >= AMQP_VERSION_CODE(0, 10, 0, 1)
```

Comment:
according to this: https://github.com/alanxz/rabbitmq-c/blob/master/librabbitmq/amqp_ssl_socket.h#L68
it is supported since v0.9.0

Reply:
It actually is 0.10.0, the comment there is wrong.

Check the [Changelog](https://github.com/alanxz/rabbitmq-c/releases/tag/v0.10.0). The [commit](https://github.com/alanxz/rabbitmq-c/commit/60adf5f8817f32b25a99aa54ba186a29208c7942) responsible for adding that method is marked as being present in 0.10.0 and master only.